exporter/collector/logs.go

```diff
@@ -435,22 +439,32 @@ func (l logMapper) logToSplitEntries(
 		delete(attrsMap, HTTPRequestAttributeKey)
 	}
 
-	if log.SeverityNumber() < 0 || int(log.SeverityNumber()) > len(severityMapping)-1 {
-		return []logging.Entry{entry}, fmt.Errorf("unknown SeverityNumber %v", log.SeverityNumber())
+	if logRecord.SeverityNumber() < 0 || int(logRecord.SeverityNumber()) > len(severityMapping)-1 {
+		return []logging.Entry{entry}, fmt.Errorf("unknown SeverityNumber %v", logRecord.SeverityNumber())
 	}
-	severityNumber := log.SeverityNumber()
+	severityNumber := logRecord.SeverityNumber()
 	// Log severity levels are based on numerical values defined by Otel/GCP, which are informally mapped to generic text values such as "ALERT", "Debug", etc.
 	// In some cases, a SeverityText value can be automatically mapped to a matching SeverityNumber.
 	// If not (for example, when directly setting the SeverityText on a Log entry with the Transform processor), then the
 	// SeverityText might be something like "ALERT" while the SeverityNumber is still "0".
 	// In this case, we will attempt to map the text ourselves to one of the defined Otel SeverityNumbers.
 	// We do this by checking that the SeverityText is NOT "default" (ie, it exists in our map) and that the SeverityNumber IS "0".
 	// (This also excludes other unknown/custom severity text values, which may have user-defined mappings in the collector)
-	if severityForText, ok := otelSeverityForText[strings.ToLower(log.SeverityText())]; ok && severityNumber == 0 {
+	if severityForText, ok := otelSeverityForText[strings.ToLower(logRecord.SeverityText())]; ok && severityNumber == 0 {
 		severityNumber = severityForText
 	}
 	entry.Severity = severityMapping[severityNumber]
 
+	// Parse severityNumber > 17 (error) to a GCP Error Reporting entry if enabled
+	if severityNumber >= 17 && l.cfg.LogConfig.ErrorReportingType {
+		if logRecord.Body().Type() != pcommon.ValueTypeMap {
+			strValue := logRecord.Body().AsString()
+			logRecord.Body().SetEmptyMap()
+			logRecord.Body().Map().PutStr("message", strValue)
+		}
+		logRecord.Body().Map().PutStr("@type", "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent")
```

Comment:
I think making this type String to be a constant might be better for readability. But I do not have a strong preference here since this is not used anywhere else.

Reply:
I noticed we are using this field in test as well, I would still defer the decision to you, but I now have a slight preference to making this a const field.